packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -492,20 +492,38 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
   }
 }
 
-/// Rounded rectangle surface that looks like an iOS popup surface, e.g., alert dialog
-/// and action sheet.
+/// An iOS-style rounded rectangle surface useful for showing modal overlays,
+/// such as dialogs and action sheets.
 ///
-/// A [CupertinoPopupSurface] can be configured to paint or not paint a white
-/// color on top of its blurred area. Typical usage should paint white on top
-/// of the blur. However, the white paint can be disabled for the purpose of
-/// rendering divider gaps for a more complicated layout, e.g., [CupertinoAlertDialog].
-/// Additionally, the white paint can be disabled to render a blurred rounded
-/// rectangle without any color (similar to iOS's volume control popup).
+/// By default, this surface is composed of a transparent surface color painted
+/// atop a [BackdropFilter]. The surface color is light gray when the ambient
+/// [CupertinoTheme] brightness is [Brightness.light], and dark gray when the
+/// ambient brightness is [Brightness.dark]. The [BackdropFilter] applied
+/// beneath the surface saturates and blurs underlying content.
+///
+/// The blur strength can be changed by setting [blurSigma] to a positive value,
+/// or removed by setting the [blurSigma] to 0.
+///
+/// The saturation effect can be removed by setting [isVibrancePainted] to
+/// false. Removing the vibrance can prevent oversaturating areas where multiple
+/// [CupertinoPopupSurface]s are stacked upon eachother.
+///
+/// For more complicated layouts, the surface color can be disabled by setting
+/// [isSurfacePainted] to false. This can be used to render divider gaps, such
+/// as with [CupertinoAlertDialog], or for rendering custom surface colors.
+///
+/// CupertinoPopupSurface uses ImageFilter.compose, which applies an inner
+/// filter first, followed by an outer filter (e.g. result =
+/// outer(inner(source))).
+///
+/// For CupertinoPopupSurface, the pixels underlying the surface are first
+/// saturated with a ColorFilter, and the resulting saturated pixels are blurred
+/// with ImageFilter.blur.
```

Comment:
As suggested above, the two paragraphs can be moved up and reduced to one sentence.

Reply:
Added